fundingmanager.go

```diff
@@ -1251,6 +1254,46 @@ func (f *fundingManager) announceChannel(localIDKey, remoteIDKey, localFundingKe
 	f.cfg.SendAnnouncement(ann.chanAnn)
 	f.cfg.SendAnnouncement(ann.chanUpdateAnn)
 	f.cfg.SendAnnouncement(ann.chanProof)
+
+	// Now that the channel is announced to the network, we will also create
+	// and send a node announcement. This is done since a node announcement
+	// is only accepted after a channel is known for that particular node,
+	// and this might be our first channel.
+	graph := f.cfg.Wallet.ChannelDB.ChannelGraph()
+	self, err := graph.FetchLightningNode(f.cfg.IDKey)
```

Comment:
The `graph.SourceNode` method can be used in place of `FetchLightningNode`. Worth noting though that I don't consider this a blocker to merging the PR.